blockonomics-woocommerce.php

```diff
@@ -197,8 +191,21 @@ function value_changed() {
                 document.getElementById('blockonomics_api_updated').value = 'true';
             }
             function checkForAPIKeyChange() {
+                let apiKey = "<?php echo get_option("blockonomics_api_key")?>";
+                //Settings have changed, click on Save first
                 if (document.getElementById('blockonomics_api_updated').value == 'true') {
                     alert('Settings have changed, click on Save first');
+                } else if (apiKey && apiKey.length === 43){
+                    document.testSetupForm.submit();
+                } else {
+                    RunTests_APIKey_Not_Set();
+                }
+            }
+            //If BCH enabled give error --- Otherwise, run test Setup for BTC
+            function RunTests_APIKey_Not_Set() {
+                let BCH_Enabled = "<?php echo get_option("blockonomics_bch")?>";
+                if(BCH_Enabled === "1"){
+                    alert("Set the API Key or disable BCH");
```

Comment:
This should rather be moved into testSetup() directly. The logic should be placed there as it will make this functionality easier to port to other integrations. Some integrations do not include the javascript to checkForAPIKeyChange.

Reply:
understood, moved the logic to testSetup();